src/config.rs

```diff
@@ -226,30 +277,30 @@ enum Config {
 }
 
 impl Config {
-    fn as_environment(self) -> Option<Environment> {
+    fn as_environment(&self) -> Option<Environment> {
         match self {
-            Config::Environment(e) => Some(e),
+            Config::Environment(e) => Some((*e).clone()),
             _ => None,
         }
     }
 
-    fn as_environment_type(self) -> Option<EnvironmentType> {
+    fn as_environment_type(&self) -> Option<EnvironmentType> {
         match self {
-            Config::EnvironmentType(e) => Some(e),
+            Config::EnvironmentType(e) => Some((*e).clone()),
```

Comment:
Why do we need to change these?

Reply:
Was recommended by clippy

Reply:
Right. Clippy says we should not be using a `&self` as a parameter. Changing our self to a ref prevented us from being able to return a non-reference.

Reply:
I'll take a look, but perhaps it's as simple as a bad naming convention for the method

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention implies that we can rename these to into_foo instead